url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/sign/AbstractModel.java

```diff
@@ -0,0 +1,94 @@
+package org.agetac.common.model.sign;
+
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.impl.Position;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public abstract class AbstractModel implements IJsonable, IModel {
+
+	protected String uniqueID;
+	protected String nom;
+	protected Position position;
+
+	public AbstractModel(String nom, Position position){
+		if (nom == null) {
+			nom = "";
+		}
+		if (position == null) {
+			position = new Position(0, 0);
+		}
+		
+		this.nom = nom;
+		this.position = position;
+	}
+	
+	public AbstractModel(String uid, String nom, Position position) {
+		this(nom, position);
+		if (uid == null) {
+			uid = "";
+		}
+		this.uniqueID = uid;
+	}
+
+	public AbstractModel(JSONObject json) throws InvalidJSONException {
+		try {
+			this.uniqueID = json.getString("uniqueID");
+			this.nom = json.getString("nom");
+			this.position = new Position(json.getJSONObject("position"));
+		} catch (JSONException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)